common/src/main/java/com/microsoft/identity/common/CodeMarkerManager.java

```diff
@@ -0,0 +1,109 @@
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common;
+
+import android.os.Environment;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class CodeMarkerManager {
+
+    private static boolean enableCodeMarker = false;
+    private static int MAX_SIZE_CODE_MARKER = 1000;
+    private static volatile List<CodeMarker> codeMarkers = new ArrayList<CodeMarker>();
+    private static long baseMilliSeconds = 0;
+    private static String scenarioCode = "";
+
+    public static void setPrefixScenarioCode(String scenarioCode) {
+        CodeMarkerManager.scenarioCode = scenarioCode;
+    }
+
+    public static void markCode(String marker) {
+        if(enableCodeMarker) {
+            if(codeMarkers.size() >= MAX_SIZE_CODE_MARKER) {
+                clearMarkers();
+            }
+            long currentMilliSeconds = System.currentTimeMillis();
+            if (codeMarkers.size() == 0) {
+                baseMilliSeconds = currentMilliSeconds;
+            }
+            SimpleDateFormat f = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS");
+            codeMarkers.add(new CodeMarker(scenarioCode + marker, currentMilliSeconds - baseMilliSeconds, f.format(new Date()), Thread.currentThread().getId()));
+        }
+    }
+
+    public static void setEnableCodeMarker(boolean enableCodeMarker) {
+        CodeMarkerManager.enableCodeMarker = enableCodeMarker;
+    }
+
+    public static void clearMarkers(){
+        codeMarkers.clear();
+    }
+
+    public static void clearAll() {
+        codeMarkers.clear();
+        CodeMarkerManager.scenarioCode = "";
+    }
+
+    public static void writeToFile(final String fileName) {
+        String stringToWrite = getFileContent();
+        File file = Environment.getExternalStorageDirectory();
+        String strFilePath = file.getAbsolutePath() + "/" + fileName;
+        try {
+            FileOutputStream fileOutputStream = new FileOutputStream(strFilePath);
+            FileWriter fileWriter = new FileWriter(fileOutputStream.getFD());
+            fileWriter.write(stringToWrite);
+            fileWriter.close();
+            fileOutputStream.getFD().sync();
+            fileOutputStream.close();
+            System.out.println("Test");
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
```

Comment:
Instead of printing stack trace, you probably want to log an error statement here

Reply:
This method is no longer part of this class now as file writing will be done by FileAppender.